internal/hud/server/websocket_reader_test.go

```diff
@@ -149,8 +150,27 @@ func (f *websocketReaderFixture) assertHandlerCallCount(n int) {
 }
 
 func (f *websocketReaderFixture) assertLogs(msg string) {
-	logs := f.out.String()
-	assert.Contains(f.t, logs, msg)
+	ctx, cancel := context.WithTimeout(f.ctx, time.Millisecond*10)
+	defer cancel()
+	isCanceled := false
+
+	for {
+		logs := f.out.String()
```

Comment:
you might need to use a ThreadSafeBuffer, to avoid weird race conditions with corrupt memory? it might even make sense to make this timeout loop a helper function on ThreadSafeBuffer

Reply:
ooh i didn't realize we had this already, awesome!